python/cudf/cudf/core/udf/strings_lowering.py

```diff
@@ -50,6 +51,41 @@ def masked_len_impl(context, builder, sig, args):
     return ret._getvalue()
 
 
+@cuda_lower(
+    "MaskedType.replace",
+    MaskedType(string_view),
+    MaskedType(string_view),
+    MaskedType(string_view),
+)
+def masked_string_view_replace_impl(context, builder, sig, args):
+    ret = cgutils.create_struct_proxy(sig.return_type)(context, builder)
+
+    src_masked = cgutils.create_struct_proxy(sig.args[0])(
+        context, builder, value=args[0]
+    )
+    to_replace_masked = cgutils.create_struct_proxy(sig.args[0])(
+        context, builder, value=args[1]
+    )
+    replacement_masked = cgutils.create_struct_proxy(sig.args[0])(
+        context, builder, value=args[2]
+    )
```

Comment:
Some of the boilerplatey-ness here could be reduced with something like:

```python
src_masked, to_replace_masked, replacement_masked = create_struct_proxies(context, builder, sig.args[0], args)
```

Reply:
something like https://github.com/rapidsai/cudf/pull/12207/commits/51ba4e3b738c0939ea723bebf4fbc13a22fa9790 ?